bigquery_etl/shredder/delete.py

```diff
@@ -223,30 +224,76 @@ def create_job(client):
 
 def get_partition_expr(table):
     """Get the SQL expression to use for a table's partitioning field."""
-    for field in table.schema:
-        if field.name != table.time_partitioning.field:
-            continue
-        if field.field_type == "TIMESTAMP":
-            return f"DATE({field.name})"
-        return field.name
+    if table.range_partitioning:
+        return table.range_partitioning.field
+    if table.time_partitioning:
+        return next(
```

Comment:
Am I understanding correctly the intention here is that `next` returns the first item from the expression? It would feel more idiomatic to use a list comprehension and take the first element via `[0]`.

Reply:
`next` is more performant, because it stops after the iterator returns one thing. I would argue that it's more idiomatic to use `next`:
```
Help on built-in function next in module builtins:

next(...)
    next(iterator[, default])

    Return the next item from the iterator. If default is given and the iterator
    is exhausted, it is returned instead of raising StopIteration.
```

Reply:
irrelevant, because we don't need to iterate, we can use: `f"CAST({table.time_partitioning.field} AS DATE)"`